src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -283,12 +299,29 @@ def set_key(cmd,
             raise CLIError(str(exception))
 
         if retrieved_kv is None:
+            if content_type and __is_json_content_type(content_type):
+                try:
+                    # Ensure that provided value is valid JSON. Error out if value is invalid JSON.
+                    value = '""' if value is None else value
+                    value = json.dumps(json.loads(value), separators=(',', ':'))
```

Comment:
From CLI, it is different for value to be null or empty string. As value is in request json payload backend can tell the difference.

Reply:
I tried "appconfig kv set --key nullValue -n mystore" and it stored the value as null instead of empty string. 

---
In reply to: [451039877](https://github.com/Azure/azure-cli/pull/14170#discussion_r451039877) [](ancestors = 451039877,448768628)